frontend/app/services/cache-service.js

```diff
@@ -0,0 +1,77 @@
+//-- copyright
+// OpenProject is a project management system.
+// Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License version 3.
+//
+// OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+// Copyright (C) 2006-2013 Jean-Philippe Lang
+// Copyright (C) 2010-2013 the ChiliProject Team
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+//
+// See doc/COPYRIGHT.rdoc for more details.
+//++
+
+module.exports = function(
+  HALAPIResource,
+  $http,
+  $q,
+  CacheFactory) {
+
+  var cacheName = 'openproject-cache';
+  var _cache = CacheFactory(cacheName, {
+    maxAge: 30 * 60 * 1000, // 30 mins
+    storageMode: 'sessionStorage'
```

Comment:
After reading a little more stuff up and getting other stuff out of my memory:

I would have expected either `memory` or `localStorage` here:

Memory is the least persistent one, a page refresh will clear it. This is the least invasive option, because good old F5 will get me to a well-known state.

If we want to cache across refresh-cycles we would need one of the other storage options. I (personally) did not know the difference between sessionStorage and localStorage, just knowing the latter one. If I understand the difference correctly, the localStorage will work browser wide, while the session storage will work "per window". If that is the only difference, I would suggest using the localStorage rather than the session storage, since this will also cache across multiple tabs.

Still my favorite for now is the memory based storage, since it is the least invasive one and it still works well if our application is used one-page-style.